src/utils_dpdk.c

```diff
@@ -705,10 +727,12 @@ static void dpdk_helper_check_pipe(dpdk_helper_ctx_t *phc) {
     }
   }
   while (data_avail) {
-    int nbytes = read(phc->pipes[0], buf, sizeof(buf));
+    int nbytes = read(phc->pipes[0], buf, (sizeof(buf) - 1));
+    DEBUG("%s:dpdk_helper_check_pipe: read nbytes=%d", phc->shm_name, nbytes);
     if (nbytes <= 0)
       break;
-    sstrncpy(out, buf, nbytes);
+    buf[nbytes] = '\n';
```

Comment:
Why `\n`? Did you intend to nil-terminate the string? In that case use `buf[nbytes] = 0;` instead.

Reply:
It is just defined char. sstrncpy ignore char at that position anyway.  Nevertheless you are right  '\0' will be better here.